python/ml4ir/base/model/calibration/temperature_scaling.py

```diff
@@ -324,6 +326,47 @@ def nll_with_lbfgs():
                      f'test accuracy: {acc_test_temperature_scaling} \n')
 
     # the file is big and should be zipped
-    zip_dir_path = dict_to_zipped_csv(data_dic, logs_dir_local, file_io)
-    logger.info(f"Read and zipped  {zip_dir_path}")
+    #zip_dir_path = dict_to_zipped_csv(data_dic, logs_dir_local, file_io)
+    #logger.info(f"Read and zipped  {zip_dir_path}")
```

Comment:
takes too long to zip.. commenting for now

Reply:
Fair. But you should be trying with smaller data instead! 

Tip: you should start with a test for this functionality (you know we will ask for a test ;-)). This test will take you some time to write (but you will commit it too) and it will check that the predictions are the same before saving and after saving. Since you are training every time set the seeds and use one epoch. Also, this way you will be able to develop locally (I hope that you are prototyping this locally and don't use splunk to debug as it is a huge pain). There already temp data for the test..

Reply:
yes It works nice in my local box and with small data. I had added the test for zipping part before . See [here.](https://github.com/salesforce/ml4ir/blob/master/python/ml4ir/applications/classification/tests/test_calibration.py#L17)
Apparently, it gets stuck with real (big) data!